tiledb/sm/query/result_tile.cc

```diff
@@ -254,9 +254,10 @@ Status ResultTile::read(
     auto buff = static_cast<unsigned char*>(buffer);
     auto cell_size = coords_tile_.first.cell_size();
     auto dim_size = cell_size / domain_->dim_num();
-    uint64_t offset = dim_size * (dim_offset);
+    uint64_t offset = dim_size * (dim_offset) * (pos + 1);
```

Comment:
It should be:
`uint64_t offset = pos * cell_size + dim_offset * dim_size;`

That is, first move to the beginning of the `pos`-th zipped cell, then move to the `dim_offset`-th coordinate in that cell.